modules/flake-parts/nixosConfigurations.nix

```diff
@@ -0,0 +1,177 @@
+{ self, lib, inputs, ... }: {
+  flake.nixosConfigurations = let
+    nixosSystem = args:
+      # TODO: flake-parts does not expose lib.nixosSystems.
+      #   Fix this upstream at flake-parts or nixpkgs.
+      #   (Why are there even two different libs ?)
+      import (inputs.nixpkgs + /nixos/lib/eval-config.nix) (
+        args // {
+          modules = args.modules ++ [{
+            system.nixos.versionSuffix =
+              ".${lib.substring 0 8 (self.lastModifiedDate or self.lastModified or "19700101")}.${self.shortRev or "dirty"}";
+            system.nixos.revision = lib.mkIf (self ? rev) self.rev;
+          }];
+        } // lib.optionalAttrs (! args?system) {
+          # Allow system to be set modularly in nixpkgs.system.
+          # We set it to null, to remove the "legacy" entrypoint's
+          # non-hermetic default.
+          system = null;
+        }
+      );
+    fake-hardware = {
+      boot.loader.grub.devices = [ "/dev/sda" ];
+      fileSystems."/" = {
+        device = "/dev/sda";
+      };
+    };
+    # some example configuration to make it eval
+    dummy = { config, ... }: {
+      networking.hostName = "example-common";
+      system.stateVersion = config.system.nixos.version;
+      users.users.root.initialPassword = "fnord23";
+    };
+  in
+  {
+    # General
+    example-common = nixosSystem {
+      system = "x86_64-linux";
+      modules = [
+        dummy
+        fake-hardware
+        self.nixosModules.common
+      ];
+    };
```

Comment:
It would be cool if those could live near their respective modules.

Reply:
We could make each module a directory and put an `example.nix` inside it.